Source/WTF/wtf/ContinuousArenaMalloc.cpp

```diff
@@ -122,19 +122,29 @@ void ContinuousArenaMalloc::internalFree(void* ptr)
     dallocx(ptr, MALLOCX_TCACHE_NONE);
 }
 
-bool ContinuousArenaMalloc::isValidRange(void *addr,
-                                        size_t size)
+bool ContinuousArenaMalloc::isValidRange(void *addr, size_t size)
 {
     ASSERT(s_Mutex->tryLock() == false);
 
     ASSERT(s_Start != NULL);
-    ASSERT(s_End != NULL);
-    ASSERT(s_Current != NULL);
+    ASSERT(s_Current >= s_Start);
+    ASSERT(s_End >= s_Current);
 
-    return (s_Current >= s_Start
-            && s_Current <= s_End
-            && s_Current + size >= s_Start
-            && s_Current + size <= s_End);
+    char* start = reinterpret_cast<char*>(addr);
```

Comment:
I notice this file is sometimes `char *blah` and `char* blah`. Bitter experience has taught me to prefer `char *blah` :)